typescript/sdk/src/providers/SmartProvider/SmartProvider.ts

```diff
@@ -310,10 +341,21 @@ export class HyperlaneSmartProvider
               ...providerMetadata,
             },
             `Error from provider.`,
-            isLastProvider ? '' : 'Triggering next provider.',
           );
           providerResultErrors.push(result.error);
-          pIndex += 1;
+
+          const nonRetryable = hasNonRetryableError({
+            method,
+            error: result.error as Error,
+          });
+
+          pIndex += nonRetryable
+            ? providers.length // Setting this to providers.length will force exit the loop and not retry with the next provider
```

Comment:
feels a little bit like a hack, wonder if there's a nicer way to do it?

worried that it's so dependent on the `if (pIndex < providers.length) {` and if that were to change, it's not obvious to change it here too

Reply:
good point. I've changed this to instead throw